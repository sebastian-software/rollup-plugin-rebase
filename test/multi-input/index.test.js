import { bundleWithMultiInput, clean, list, read } from "../util"
import { join, sep } from "path"

const root = __dirname

test("Multi input support", async () => {
  await bundleWithMultiInput(root, ['src/**/!(*.test).js'], "output", {}, {
    transformOutputPath: (output) => {
      // Need to transform output as bundle path starts from root of project
      return output.replace(`${join('test', 'multi-input', 'src')}${sep}`, '');
    },
  })

  expect(await list(root, "output")).toMatchSnapshot()
  expect(await read(root, "output/index.js")).toMatchSnapshot()
  expect(await read(root, "output/sub/folder/child.js")).toMatchSnapshot()

  await clean(root, "output")
})
