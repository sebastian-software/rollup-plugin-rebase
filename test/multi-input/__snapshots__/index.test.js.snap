// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multi input support 1`] = `
"HDWIGpiP.md
SmVOQyTt.png
index.js
sub"
`;

exports[`Multi input support 2`] = `
"import index from './sub/folder/child.js';
export { default } from './sub/folder/child.js';
import './SmVOQyTt.png';
import './HDWIGpiP.md';
"
`;

exports[`Multi input support 3`] = `
"import image from '../../SmVOQyTt.png';
import content from '../../HDWIGpiP.md';

function index() {
  return content + image
}

export { index as default };
"
`;
